tests/scripts/generate_test_code.py

```diff
@@ -0,0 +1,722 @@
+                     dispatching test cases from .data file.
+helper .function - Read common reusable functions.
+"""
+
+
+import io
+import os
+import re
+import sys
+import argparse
+import shutil
+
+
+BEGIN_HEADER_REGEX = '/\*\s*BEGIN_HEADER\s*\*/'
+END_HEADER_REGEX = '/\*\s*END_HEADER\s*\*/'
+
+BEGIN_SUITE_HELPERS_REGEX = '/\*\s*BEGIN_SUITE_HELPERS\s*\*/'
+END_SUITE_HELPERS_REGEX = '/\*\s*END_SUITE_HELPERS\s*\*/'
+
+BEGIN_DEP_REGEX = 'BEGIN_DEPENDENCIES'
+END_DEP_REGEX = 'END_DEPENDENCIES'
+
+BEGIN_CASE_REGEX = '/\*\s*BEGIN_CASE\s*(.*?)\s*\*/'
+END_CASE_REGEX = '/\*\s*END_CASE\s*\*/'
+
+
+class InvalidFileFormat(Exception):
+    """
+    Exception to indicate invalid file format.
+    """
+    pass
+
+
+class FileWrapper(io.FileIO):
+    """
+    File wrapper class. Provides reading with line no. tracking.
+    """
+
+    def __init__(self, file_name):
+        """
+        Init file handle.
+
+        :param file_name: File path to open.
+        """
+        super(FileWrapper, self).__init__(file_name, 'r')
+        self.line_no = 0
+
+    def __next__(self):
+        """
+        Iterator return impl.
+        :return: Line read from file.
+        """
+        line = super(FileWrapper, self).__next__()
+        if line:
+            self.line_no += 1
+            # Convert byte array to string with correct encoding
+            return line.decode(sys.getdefaultencoding())
+        return None
+
+
+def split_dep(dep):
+    """
+    Split NOT character '!' from dependency. Used by gen_deps()
+
+    :param dep: Dependency list
+    :return: list of tuples where index 0 has '!' if there was a '!' before the dependency string
+    """
+    return ('!', dep[1:]) if dep[0] == '!' else ('', dep)
+
+
+def gen_deps(deps):
+    """
+    Generates dependency i.e. if def and endif code
+
+    :param deps: List of dependencies.
+    :return: if defined and endif code with macro annotations for readability.
+    """
+    dep_start = ''.join(['#if %sdefined(%s)\n' % split_dep(x) for x in deps])
+    dep_end = ''.join(['#endif /* %s */\n' % x for x in reversed(deps)])
+
+    return dep_start, dep_end
+
+
+def gen_deps_one_line(deps):
+    """
+    Generates dependency checks in one line. Useful for writing code in #else case.
+
+    :param deps: List of dependencies.
+    :return: ifdef code
+    """
+    defines = ('#if ' if len(deps) else '') + ' && '.join(['%sdefined(%s)' % split_dep(x) for x in deps])
+    return defines
+
+
+def gen_function_wrapper(name, locals, args_dispatch):
+    """
+    Creates test function wrapper code. A wrapper has the code to unpack parameters from parameters[] array.
+
+    :param name: Test function name
+    :param locals: Local variables declaration code
+    :param args_dispatch: List of dispatch arguments. Ex: ['(char *)params[0]', '*((int *)params[1])']
+    :return: Test function wrapper.
+    """
+    # Then create the wrapper
+    wrapper = '''
+void {name}_wrapper( void ** params )
+{{
+    {unused_params}
```

Comment:
When `{unused_params}` is empty, this generates a line with whitespace which `check-files.py` complains about. Although generated code could be exempt from `check-files.py`, it's more convenient that it passes, otherwise this creates noise if we happen to run `check-files` after generating test code, which is common on developer machines and could well happen on the CI.

Suggested patch which also improves the Python style:
```
--- a/tests/scripts/generate_test_code.py
+++ b/tests/scripts/generate_test_code.py
@@ -136,11 +136,11 @@ def gen_function_wrapper(name, locals, args_dispatch):
     wrapper = '''
 void {name}_wrapper( void ** params )
 {{
-    {unused_params}
-{locals}
+{unused_params}{locals}
     {name}( {args} );
 }}
-'''.format(name=name, unused_params='(void)params;' if len(args_dispatch) == 0 else '',
+'''.format(name=name,
+           unused_params='' if args_dispatch else '    (void) params;\n',
            args=', '.join(args_dispatch),
            locals=locals)
     return wrapper
```